web/src/people.ts

```diff
@@ -757,15 +758,24 @@ export function format_small_avatar_url(raw_url: string): string {
 
 export function sender_is_bot(message: Message): boolean {
     if (message.sender_id) {
-        const person = get_by_user_id(message.sender_id);
+        const ignore_missing = !settings_data.user_can_access_all_other_users();
+        const person = maybe_get_user_by_id(message.sender_id, ignore_missing);
+        if (person === undefined) {
+            return false;
+        }
         return person.is_bot;
     }
     return false;
 }
 
 export function sender_is_guest(message: Message): boolean {
     if (message.sender_id) {
-        const person = get_by_user_id(message.sender_id);
+        const ignore_missing = !settings_data.user_can_access_all_other_users();
```

Comment:
We might also set `ignore_missing` as `True` for all cases and assume that the user ID would be valid. Anyway here as well we are atleast assuming that for guests and when the user access setting is set to "members".

Reply:
OK -- let's add a comment to that effect that we'll find if grepping for the setting name, as that's a thing we'll certainly do if changing the permissions options.